internal/data/resource_repository.go

```diff
@@ -211,7 +211,6 @@ func (r *resourceRepository) FindResourceByKeys(tx *gorm.DB, key bizmodel.Report
 	query = query.Where("LOWER(rr.local_resource_id) = LOWER(?)", key.LocalResourceId().Serialize())
 	query = query.Where("LOWER(rr.resource_type) = LOWER(?)", key.ResourceType().Serialize())
 	query = query.Where("LOWER(rr.reporter_type) = LOWER(?)", key.ReporterType().Serialize())
-	query = query.Where("rr.tombstone = ?", false)
```

Comment:
I think removing this will cause consecutive delete calls to the same resource to return a 200 and increment `representation_version` arbitrarily even when it is already tombstoned.

1. Report Resource (version 0, tombstone=f)
2. Delete Resource (version 1, tombstone=t)
3. Delete Resource (version 2, tombstone=t)

Reply:
Yeah, but the issue is if I do not exclude soft deleted resource, it will create a new resource and a follow on delete will fail the unique constraint because it will look exactly the same as the previous delete. That was the issue to start with.

Reply:
I do have to test with check and check for update with this removal, adding tests for that now.